content/riak/kv/2.1.4/using/cluster-operations/reindexing-data.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "Reindexing Data"
+description: ""
+project: "riak_kv"
+project_version: "2.1.4"
+menu:
+  riak_kv-2.1.4:
+    name: "Reindexing Data"
+    identifier: "cluster_operations_reindex_data"
+    weight: 118
+    parent: "managing_cluster_operations"
+toc: true
+canonical_link: "https://docs.basho.com/riak/kv/latest/using/cluster-operations/reindexing-data"
+---
+
+After making [changes to your search schema](../updating-search-schemas) we recommend reindexing existing search data. You can reindex data by either:
+
+- [Creating a script to 're-PUT' all of the data](#reindex-data-with-custom-script).
+- [Using the Riak Data Migrator](#reindex-data-with-riak-data-migrator)
+- [Rebuilding Yokozuna Active Anty Entropy (AAE) trees](#reindex-data-with-aae-rebuild)
+
+## Reindexing Data With Custom Script
+
+You can reindex existing data by creating a script that 're-PUTs' all of your data. Below is an example script in <<LANGUAGE>>:
```

Comment:
What are the benefits of this vs. the others? What environment is this choice best for?